client/src/components/UploadComponent.js

```diff
@@ -0,0 +1,339 @@
+import React, { useState, useRef } from "react";
+import styled from "styled-components";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faUpload, faTrash } from "@fortawesome/free-solid-svg-icons";
+
+const KILO_BYTES_PER_BYTE = 1000;
+const DEFAULT_MAX_FILE_SIZE_IN_BYTES = 500000;
+
+const convertNestedObjectToArray = (nestedObj) =>
+  Object.keys(nestedObj).map((key) => nestedObj[key]);
+
+const convertBytesToKB = (bytes) => Math.round(bytes / KILO_BYTES_PER_BYTE);
+
+const FileUpload = ({
+  label,
+  updateFilesCb,
+  maxFileSizeInBytes = DEFAULT_MAX_FILE_SIZE_IN_BYTES,
+  ...otherProps
+}) => {
+  const fileInputField = useRef(null);
+  const [files, setFiles] = useState({});
+
+  const handleUploadBtnClick = () => {
+    fileInputField.current.click();
+  };
+
+  const addNewFiles = (newFiles) => {
+    for (let file of newFiles) {
+      if (file.size < maxFileSizeInBytes) {
+        if (!otherProps.multiple) {
+          return { file };
+        }
+        files[file.name] = file;
+      }
+    }
+    return { ...files };
+  };
+
+  const callUpdateFilesCb = (files) => {
+    const filesAsArray = convertNestedObjectToArray(files);
+    updateFilesCb(filesAsArray);
+  };
+
+  const handleNewFileUpload = (e) => {
+    const { files: newFiles } = e.target;
+    if (newFiles.length) {
+      let updatedFiles = addNewFiles(newFiles);
+      setFiles(updatedFiles);
+      callUpdateFilesCb(updatedFiles);
+    }
+  };
+
+  const removeFile = (fileName) => {
+    delete files[fileName];
+    setFiles({ ...files });
+    callUpdateFilesCb({ ...files });
+  };
+
+  return (
+    <>
+      <FileUploadContainer>
```

Comment:
Remove the React fragments as the <FileUploadContainer> already wraps the whole content